templates/aws-stack.yml

```diff
@@ -939,6 +939,59 @@ Resources:
       AutoScalingReplacingUpdate:
         WillReplace: true
 
+  AsgProcessSuspenderRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+        - Action: ['sts:AssumeRole']
+          Effect: Allow
+          Principal:
+            Service: [lambda.amazonaws.com]
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+      Policies:
+      - PolicyName: AsgProcessModification
+        PolicyDocument:
+          Version: 2012-10-17
+          Statement:
+          - Effect: Allow
+            Action:
+            - 'autoscaling:SuspendProcesses'
+            Resource: '*'
```

Comment:
is this OK? Is there a way to only give permission to the specific ASG in this stack?

Reply:
I would think it should be possible!

Reply:
I tried this:

```diff
diff --git a/templates/aws-stack.yml b/templates/aws-stack.yml
index 40490c9..a8de337 100644
--- a/templates/aws-stack.yml
+++ b/templates/aws-stack.yml
@@ -959,7 +959,7 @@ Resources:
           - Effect: Allow
             Action:
             - 'autoscaling:SuspendProcesses'
-            Resource: '*'
+            Resource: !GetAtt AgentAutoScaleGroup.Arn
```


.. but sadly it seems the `AWS::AutoScaling::AutoScalingGroup` type doesn't have an Arn property we can get:


```
$ STACK_NAME=jh-azrebalance7 aws-vault exec bk-sandbox-admin make create-stack      
...

An error occurred (ValidationError) when calling the CreateStack operation: Template error: resource AgentAutoScaleGroup does not support attribute type Arn in Fn::GetAtt
make: *** [Makefile:114: create-stack] Error 255
```

Reply:
https://github.com/aws-cloudformation/aws-cloudformation-coverage-roadmap/issues/548

> I would also like to see this implemented so we can write IAM policies that limit access to a specific autoscaling group - otherwise, there is no way to target a specific group (as far as I can tell there is no way to get the UUID part of the group ARN and IAM won't take a * there).

.. and then:

> Correction - Targetting specific autoscaling groups by their "friendly name" can work if we use wildcards in place of the region and account id (instead of just leaving these empty - like we do with other resources such as S3 objects). To this does work: `!Sub arn:aws:autoscaling:*:*:autoScalingGroup:*:autoScalingGroupName/${LogicalGroupName}`

Reply:
Huh yeah, nothing but “friendly name”:

![image](https://user-images.githubusercontent.com/15759/96809635-00a60780-1467-11eb-89fa-96947f81353a.png)

The wildcard approach is pretty legit, but you can be specific on region & account:

```
Resource: !Sub arn:aws: autoscaling:${AWS::Region}:${AWS::AccountId}:autoScalingGroup:…
```